shub/utils.py

```diff
@@ -27,8 +27,11 @@
 # https://github.com/scrapinghub/shub/pull/309#pullrequestreview-113977920
 try:
     from pip import main as pip_main
-except:
-    from pip._internal import main as pip_main
+except:  # noqa
+    try:
+        from pip._internal.main import main as pip_main
+    except ModuleNotFoundError:  # noqa
+        from pip._internal import main as pip_main
```

Comment:
@vshlapakov this above changes fixes this error on py38: 
https://ci.appveyor.com/project/scrapinghub/shub/builds/33969869/job/veplrw43tgu513o4

Reference:
https://stackoverflow.com/questions/43398961/pip-module-has-no-attribute-main/49853826#49853826 

Now the only failing test are this on py27: 
https://ci.appveyor.com/project/scrapinghub/shub/build/job/3wp6gr9kpxgoxji5

We're almost there :)

Reply:
Cool :+1:  About the last failed test, it seems related to the recently merged https://github.com/pallets/click/pull/1591, should be fixed in the next version of click. Not sure what to do :thinking: I see it happens on Python2.7 / Windows only, so the scope is probably limited, but still it seems to be a breaking change (triggered by the first invocation of click.echo). Also, `click=8.0` is going to drop support for Python 2 anyway.

Reply:
Python 2 already hit the EoL, so I believe it should be fine to officially stop supporting it in the next minor version (`2.11`) - remove CI tests, clean setup.py etc, and do the rest of the cleanup in related PRs.

Reply:
@vshlapakov Hey Vik, my suggestion is to stick with `click==7.0` at least until we have their hotfix released, I did test with `v7.0` and it isn't throwing any error, with this we could conclude our upgrade and release the newer shub, wdyt? cya

Reply:
@rafaelcapucho I see, this is a very good point, awesome that you found it :+1: Yeah, the breaking change was introduced in 7.1 version by https://github.com/pallets/click/issues/1065 to fix specific click behavior (redirecting output to NUL on windows), let's stick with `7.0` - at least the basic functionality seems to work - and move away from Python 2 later.